<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1407</width>
    <height>887</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="controlsGroup">
      <property name="title">
       <string>Controls</string>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="cityNameLabel">
         <property name="text">
          <string>City Name:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="cityNameInput"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="srcLabel">
         <property name="text">
          <string>Source:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="srcDropdown"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="destLabel">
         <property name="text">
          <string>Destination:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="destDropdown"/>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="budgetLabel">
         <property name="text">
          <string>Budget:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="budgetInput"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="buttonLayout">
      <item>
       <widget class="QPushButton" name="addCityButton">
        <property name="text">
         <string>Add City</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addRouteButton">
        <property name="text">
         <string>Add Route</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addTripButton">
        <property name="text">
         <string>Add Trip</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="undoButton">
        <property name="text">
         <string>Undo</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="dequeueButton">
        <property name="text">
         <string>Dequeue</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="mstButton">
        <property name="text">
         <string>MST</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="searchButton">
        <property name="text">
         <string>Search</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTableWidget" name="tripTable">
      <column>
       <property name="text">
        <string>Source</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Destination</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Budget</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="pathDisplay">
      <property name="placeholderText">
       <string>MST result will be displayed here...</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="tripLog">
      <property name="placeholderText">
       <string>Trip log will show enqueue/dequeue/undo actions...</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="statusLabel">
      <property name="text">
       <string>Status messages will appear here</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
